src/analysis/retail/monk/mistweaver/modules/spells/Vivify.tsx

```diff
@@ -280,57 +275,80 @@ class Vivify extends Analyzer {
     }
   }
 
+  private _makeHealId(event: HealEvent): string {
+    return (
+      event.targetID + '_' + event.timestamp + '_' + event.amount + '_' + (event.overheal || 0)
+    );
+  }
+
   private _tallyCastEntry(vivifyHeal: HealEvent) {
     const invigoratingMistHits = getInvigHitsPerCast(vivifyHeal) as HealEvent[];
-
     this.vivifyGoodCrits = 0;
     this.vivifyWastedCrits = 0;
     let fullOverhealHits = 0;
     let healingPerCast = 0;
     let overhealPerCast = 0;
 
-    healingPerCast += vivifyHeal.amount + (vivifyHeal.absorbed || 0);
-    overhealPerCast += vivifyHeal.overheal || 0;
-    this._tallyUpliftedSpiritsCDR(vivifyHeal);
+    let rems = 0;
+    const targetsInGrouping: number[] = [];
 
+    /*Vivacious Vivification can proc during a hardcasted vivify and is not consumed until the following vivify cast. This is often
+     * causing two vivify casts and their subsequent invig mist healing to occur simultaneously. Invigorating mists can only hit a target
+     * once per vivify cast, so we can track and filter out the targets that have already been hit to properly allocate each cast.
+     */
     invigoratingMistHits.forEach((invigHeal) => {
-      const effective = invigHeal.amount + (invigHeal.absorbed || 0);
-      fullOverhealHits += effective ? 1 : 0;
-      healingPerCast += effective;
-      overhealPerCast += invigHeal.overheal || 0;
-      this._tallyUpliftedSpiritsCDR(invigHeal);
+      const targetId = invigHeal.targetID;
+      const heal_id = this._makeHealId(invigHeal);
+
+      if (
+        (!this.healsPerPlayer[targetId] || !this.healsPerPlayer[targetId].has(heal_id)) &&
+        !targetsInGrouping.find((id) => id === targetId)
+      ) {
+        const effective = invigHeal.amount + (invigHeal.absorbed || 0);
+
+        fullOverhealHits += !effective ? 1 : 0;
+        healingPerCast += effective;
+        overhealPerCast += invigHeal.overheal || 0;
+
+        this._tallyUpliftedSpiritsCDR(invigHeal);
+        this.healsPerPlayer[targetId]
+          ? this.healsPerPlayer[targetId].add(heal_id)
+          : (this.healsPerPlayer[targetId] = new Set<string>().add(heal_id));
```

Comment:
i'm not sure I follow what this `healsPerPlayer` tracking is doing. it seems like the keys will often not be duplicated because the invigorating mists heals will have different amounts/overheals even if the timestamp is identical.

actually i think i get it---this happens because the invig heal is processed twice. `targetsInGrouping` limits each vivify to processing at most 1 invig per player. then `healsPerPlayer` limits each invig to being processed at most once even if linked to multiple vivifies. is that right?

Reply:
Yep! Its a bit of a bandaid I know, if there is a cleaner way to do this let me know. The identical timestamp threw all my attempts and solving it via EventLinkNormalizer

Reply:
yea i think this would *technically* be solvable with some `additionalCondition` magic but its messy enough that I'm not sure its worth it, and it would ultimately be very similar to what you're doing here. this could use a brief explanation of what the two sets are doing but otherwise seems okay.

minor nit would be to use a `Set` instead of an array for `targetsInGrouping` but the list is small enough that I'm not sure if the overhead of using a Set is outweighed by the faster lookups

Reply:
updated